src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const cloneState = { ...state };
+  const arrayAction = [];
+
+  for (const action of actions) {
+    const { type } = action;
```

Comment:
You can get extraData and keysToRemove as well from action object

Reply:
Forgot about this...))